wire-ios-data-model/Source/Model/Connection/ZMConnection.m

```diff
@@ -117,17 +111,12 @@ - (void)setStatus:(ZMConnectionStatus)status;
         [newStatus isEqual:@(ZMConnectionStatusAccepted)]) {
         self.to.needsToBeUpdatedFromBackend = YES;
 
-        if (self.conversation.isArchived) {
+        if (self.to.oneOnOneConversation.isArchived) {
             // When a connection is accepted we always want to bring it out of the archive
-            self.conversation.isArchived = NO;
+            self.to.oneOnOneConversation.isArchived = NO;
         }
     }
 
-    if ([newStatus isEqual:@(ZMConnectionStatusCancelled)]) {
-        self.to.connection = nil;
-        self.to = nil;
-    }
-
```

Comment:
We need to maintain the relationship between user and connection since we no longer have a relation between connection and conversation.